### PR TITLE
Add v3.4.4 docs and news

### PR DESCRIPTION
**Pull request by dongjoon-hyun**
Apache Spark 3.4.4 is ready like the following.
- https://downloads.apache.org/spark/spark-3.4.4/
- https://pypi.org/project/pyspark/3.4.4/
- https://repo1.maven.org/maven2/org/apache/spark/spark-core_2.12/3.4.4/
- https://repo1.maven.org/maven2/org/apache/spark/spark-core_2.13/3.4.4/
- https://hub.docker.com/r/apache/spark/tags?name=3.4.4

Screenshot 2024-10-27 at 11 12 42
Screenshot 2024-10-27 at 11 12 25
Screenshot 2024-10-27 at 11 12 21
Screenshot 2024-10-27 at 11 13 08

### CODE REVIEW

File: site/docs/3.4.4/index.html [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+ limitations under the License.
+-->
+
+<!DOCTYPE html>
+<html lang="en-US">
+<meta charset="utf-8">
+<title>Redirecting&hellip;</title>
+<link rel="canonical" href="https://archive.apache.org/dist/spark/docs/3.4.4/">
+<script>location="https://archive.apache.org/dist/spark/docs/3.4.4/"</script>
+<meta http-equiv="refresh" content="0; url=https://archive.apache.org/dist/spark/docs/3.4.4/">
```
Comment (dongjoon-hyun) [inline site/docs/3.4.4/index.html:24]: The document is intentionally in `archive.apache.org` because this is a newly created link for the EOL release.

**Comment by dongjoon-hyun** [comment]
cc @HyukjinKwon , @cloud-fan , @gengliangwang , @MaxGekk , @zhengruifeng , @xinrong-meng , @itholic , @yaooqinn , @LuciferYang , @viirya , @huaxingao .

**Review by zhengruifeng — APPROVED** [review]

**Review by LuciferYang — APPROVED** [review]

**Review by yaooqinn — APPROVED** [review]

**Comment by dongjoon-hyun** [comment]
Thank you, @zhengruifeng , @LuciferYang , @yaooqinn .
Merged to `asf-site`.

**Comment by viirya** [comment]
Looks good to me.

**Comment by dongjoon-hyun** [comment]
Thank you, @viirya !

**Pull request merged**